src/transformers/generation/configuration_utils.py

```diff
@@ -224,19 +224,22 @@ class GenerationConfig(PushToHubMixin):
         low_memory (`bool`, *optional*):
             Switch to sequential beam search and sequential topk for contrastive search to reduce peak memory.
             Used with beam search and contrastive search.
-        watermark (`bool`, *optional*):
-            Watermark the model outputs by adding a small bias to randomly selected set of "green" tokens.
-        greenlist_ratio (`float`, *optional*):
-            Used for watermaring. The ratio of "green" tokens used to the vocabulary size. Defaults to 0.25.
-        watermark_bias (`float`, *optional*):
-            Used with watermarking. The bias added to the selected "green" tokens' logits. Defaults to 2.0.
-        hashing_key (`int`, *optional*):
-            Hahsing key used for watermarking. Defaults to 15485863 (the millionth prime).
-        seeding_scheme (`str`, *optional*):
-            Algorithm to use for watermarking. Accepts values:
-                - "lefthash" (default): "green" tokens selection depend on the last token (Algorithm 2 from paper)
-                - "selfhash": "green" tokens selection depends ono the current token itself (Algorithm 3 from paper)
-                    The downside of this scheme is that it considers all possible next tokens and can be slower than "lefthash".
+        watermarking_args (`Dict`, *optional*):
```

Comment:
Yeah, this sounds better. I also do not like it as a dict. Let me change it and test how it goes

Reply:
So the args are now in a WatermarkConfig class, which follows closely the style of other configs like "quantization_config". I had to add config's `to_dict()` and `from_dict()` calls while setting up the GenerationConfig, so that we are able to save and load it back.

Reply:
@gante Adding the new dataclass for watermark args in the same file as the Detector is causing fails in TF and FLAX tests, because of the "torch" import/usage in that file. I decided to mode it next to GenerationConfig, seems quite intuitive to me having configs in the same file. WDYT?

Reply:
Yeah, it's okay :)